### PR TITLE
Remove cold code paths

### PR DESCRIPTION
**Pull request by arianneb**
Modified function allowSplitWarmAndColdBlocks() to return true which sets the 
cold code length to 0. This ensures that the cold code paths are never taken. 
After testing with this change, it has been verified that everything works without 
the cold code paths. The cold code paths were then removed and  
allowSplitWarmAndColdBlocks() was reverted back to return false (such that any 
dependencies in openj9 may continue to use it normally, for now). 

Signed-off-by: Arianne Butler <arianne.butler@ibm.com>

[ci skip]

### CODE REVIEW

File: compiler/codegen/OMRCodeGenerator.hpp [hunk 1]
```diff
@@ -376,7 +376,7 @@ class OMR_EXTENSIBLE CodeGenerator
    bool buildInterpreterEntryPoint() { return false; }
    void generateCatchBlockBBStartPrologue(TR::Node *node, TR::Instruction *fenceInstruction) { return; }
    bool supportsUnneededLabelRemoval() { return true; }
-   bool allowSplitWarmAndColdBlocks() { return false; }
```
Comment (mgaudet) [inline compiler/codegen/OMRCodeGenerator.hpp]: Is the goal to remove this query entirely? 

I'm a little surprised to see it flip from disabled to enabled, given most places in this it's been deleted. If this is to influence a downstream project like OpenJ9, shouldn't the code be removed from OpenJ9, then subsequently this PR removes the OMR impl?

Comment (arianneb) [reply compiler/codegen/OMRCodeGenerator.hpp]: I believe the goal is to remove it, yes. There are still j9 dependencies on it. We are completing this task in 3 phases. That piece will come later once we can verify that what we've already done is kosher. @mgaudet

Comment (mgaudet) [reply compiler/codegen/OMRCodeGenerator.hpp]: I guess the thing I don't understand is why this flips sense from false to true though. Your PR message seems to suggest that you had tried to revert it, but it didn't take.

Comment (arianneb) [reply compiler/codegen/OMRCodeGenerator.hpp]: So when allowSplitWarmAndColdBlocks returns true, it sets the cold code length to 0. This ensures that the cold code paths will never be taken. I made this simple modification and ran a personal build to ensure that things will pass without using any of the cold code paths. Once that was verified, I went on to remove the cold code paths altogether and reverted allowSplitWarmAndColdBlocks to return false, such that any j9 dependencies can continue to use it for now. @mgaudet

**Comment by xliang6** [comment]
@arianneb I would ask a review from @fjeremic and @0xdaryl once you address my concerns. Thanks.

**Comment by xliang6** [comment]
LGTM.

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 2]
```diff
@@ -6496,33 +6467,8 @@ OMR::Z::CodeGenerator::doBinaryEncoding()
...
-
-   if ( !self()->allowSplitWarmAndColdBlocks() )
-      {
-      // We need this, otherwise getWarmCodeLength() would compute a -ive value,
-      // which in-turn would make the later assert check fail. The negative
-      // value comes about because getCodeStart() is larger _binaryBufferStart,
-      // which is what warmCodeEnd is initlaized to.
-      self()->setWarmCodeEnd(self()->getCodeStart());
-      }
-
```
Comment (xliang6) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: On second thought, can we get rid off code setting ```doBinaryEncoding``` to true and the ```if``` statement checking if ```doBinaryEncoding``` is true (if that is the only live code path)?

**Comment by arianneb** [comment]
I tested locally with `TR_ASSERT` enabled and by adding a `TR_ASSERT(0, ...)` in `omr/compiler/compile/OMRCompilation.cpp` in function `OMR::Compilation::Compilation(…)`. Then I ran a java -version with `-Xjit:count=0` and the program crashed as expected. I will test on the farm with `TR_ASSERT` enabled once everything is back up and running. @fjeremic @0xdaryl

**Comment by arianneb** [comment]
Commits are now split. Please append the URL of the second commit (Further cleanup of doBinaryEncoding()) with `?w=1` to ignore the white space. The un-indenting that I did makes the change difficult to read.

**Comment by 0xdaryl** [comment]
omr-genie build all

**Comment by 0xdaryl** [comment]
genie-omr build all

**Comment by arianneb** [comment]
I fixed the spelling. Testing is complete. This should be good to merge now. @0xdaryl

**Pull request merged**